File: app/src/main/java/me/devsaki/hentoid/parsers/content/ExhentaiContent.java
package me.devsaki.hentoid.parsers.content;

import android.webkit.CookieManager;

import androidx.annotation.NonNull;

import java.io.IOException;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

import me.devsaki.hentoid.database.domains.Content;
import me.devsaki.hentoid.enums.Site;
import me.devsaki.hentoid.enums.StatusContent;
import me.devsaki.hentoid.json.sources.EHentaiGalleriesMetadata;
import me.devsaki.hentoid.json.sources.EHentaiGalleryQuery;
import me.devsaki.hentoid.retrofit.sources.EHentaiServer;
import timber.log.Timber;

public class ExhentaiContent extends BaseContentParser {

    @Nullable
    public Content update(@NonNull final Content content, @Nonnull String url, boolean updateImages) {
        CookieManager mgr = CookieManager.getInstance();
        String cookiesStr = mgr.getCookie(".exhentai.org");

        String[] galleryUrlParts = url.split("/");
        if (galleryUrlParts.length > 5) {
            EHentaiGalleryQuery query = new EHentaiGalleryQuery(galleryUrlParts[4], galleryUrlParts[5]);

            try {
                EHentaiGalleriesMetadata metadata = EHentaiServer.exentaiApi.getGalleryMetadata(query, cookiesStr).execute().body();
                if (metadata != null)
                    return metadata.update(content, url, Site.EXHENTAI, updateImages);
            } catch (IOException e) {
                Timber.e(e, "Error parsing content.");
            }
        }
        return new Content().setSite(Site.EXHENTAI).setStatus(StatusContent.IGNORED);
    }
}

File: app/src/main/java/me/devsaki/hentoid/parsers/content/ExhentaiContent.kt
package me.devsaki.hentoid.parsers.content

import android.webkit.CookieManager
import me.devsaki.hentoid.database.domains.Content
import me.devsaki.hentoid.enums.Site
import me.devsaki.hentoid.enums.StatusContent
import me.devsaki.hentoid.json.sources.EHentaiGalleryQuery
import me.devsaki.hentoid.retrofit.sources.EHentaiServer
import timber.log.Timber
import java.io.IOException

class ExhentaiContent : BaseContentParser() {
    override fun update(content: Content, url: String, updateImages: Boolean): Content {
        val mgr = CookieManager.getInstance()
        val cookiesStr = mgr.getCookie(".exhentai.org")
        val galleryUrlParts = url.split("/")
        if (galleryUrlParts.size > 5) {
            val query = EHentaiGalleryQuery(galleryUrlParts[4], galleryUrlParts[5])
            try {
                val metadata =
                    EHentaiServer.exentaiApi.getGalleryMetadata(query, cookiesStr).execute().body()
                if (metadata != null) return metadata.update(
                    content,
                    url,
                    Site.EXHENTAI,
                    updateImages
                )
            } catch (e: IOException) {
                Timber.e(e, "Error parsing content.")
            }
        }
        return Content(site = Site.EXHENTAI, status = StatusContent.IGNORED)
    }
}